### PR TITLE
Save confirm improvements & Unsaved changes modal fixes

### PR DESCRIPTION
**Pull request by darcyadams**
- Pass spinnerName to confirmSave - now defaults to DataView default spinner
- `saveState` in `ConfirmSave` is now a TS enum
- unsaved changes dialogue now works

### CODE REVIEW

File: src/ConfirmSave.tsx [hunk 1]
```diff
@@ -77,6 +77,14 @@ export interface IConfirmSaveProps {
...
   displayProgressBar?: boolean;
   uploadProgress?: string | number;
+  spinnerName?: SpinnerNames;
+}
+
+export enum SaveStates {
+  Pending,
+  Complete,
+  Warning,
+  Failed
```
Comment (andyparisi) [inline src/ConfirmSave.tsx]: For enums we usually write the values in all caps

Comment (darcyadams) [reply src/ConfirmSave.tsx]: will switch it, but was copying this one...
```ts
export enum SpinnerNames {
  Circle = 'circle',
  CubeGrid = 'cube-grid',
  Wave = 'wave',
  FoldingCube = 'folding-cube',
  ThreeBounce = 'three-bounce',
  DoubleBounce = 'double-bounce',
  WanderingCubes = 'wandering-cubes',
  ChasingDots = 'chasing-dots',
  RotatingPlane = 'rotating-plane',
  Pulse = 'pulse',
  Wordpress = 'wordpress',
```

Comment (darcyadams) [reply src/ConfirmSave.tsx]: okay updated to all caps

Comment (andyparisi) [reply src/ConfirmSave.tsx]: Gotcha.. yeah I should have made those all caps that's my mistake.

**Pull request merged**